pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -468,3 +451,21 @@ func ConnectClickHouse(url string) (*sql.DB, error) {
 	}
 	return connect, nil
 }
+
+func parseDatabaseURL(url string) (protocol, string, error) {
+	match, err := regexp.MatchString(`.*://.*:(\d+)`, url)
```

Comment:
I suggest you use https://pkg.go.dev/net/url#Parse to parse the URL, then check the fields in the URL struct